eth/vm/base.py

```diff
@@ -295,6 +298,7 @@ def import_block(self, block: BlockAPI) -> BlockAPI:
         self._block = self.get_block().copy(
             header=self.configure_header(
                 coinbase=block.header.coinbase,
+                difficulty=block.header.difficulty,
```

Comment:
We currently do not pass on the `difficulty` because we rely on calculating it ourselves and I'm sure we *could*  also do that in the Clique scheme (I had that at some point) but it seems wasteful to do that here because at the end of the block import there's a call to `validate_seal` anyway at which point we'll check if the difficulty matches our expectations.

Notice that, even when we add this here, this does not meddle with the current PoW scheme which will stubbornly continue to ignore the provided difficulty and calculate it instead.

Reply:
Not actionable at this stage, but I think this suggests that we need to figure out the abstraction for some sort of generic `ConsensusAPI` or something.  Ideally, rather than ending up in this case where we're doing things to skirt around the POW stuff, we should maybe be looking at what subset of these APIs deserve to be extracted so that implementing Clique/POW/POS/etc are fully independent of each other without having to do funky stuff to compensate for parent class logic that applies to a different consensus engine.

Reply:
Yes, I agree.